src/Nethermind/Nethermind.Monitoring/MetricPusher.cs

```diff
@@ -0,0 +1,177 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#nullable enable
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Prometheus;
+
+namespace Nethermind.Monitoring;
+
+using System.Diagnostics;
+using System.Text;
+
+/// <summary>
+/// A metric server that regularly pushes metrics to a Prometheus PushGateway.
+/// </summary>
+public class MetricPusher : MetricHandler
+{
+    private readonly TimeSpan _pushInterval;
+    private readonly HttpMethod _method;
+    private readonly Uri _targetUrl;
+    private readonly Func<HttpClient> _httpClientProvider;
+    private readonly SemaphoreSlim _wait;
+
+    public MetricPusher(MetricPusherOptions options)
+    {
+        if (string.IsNullOrEmpty(options.Endpoint))
+            throw new ArgumentNullException(nameof(options.Endpoint));
+
+        if (string.IsNullOrEmpty(options.Job))
+            throw new ArgumentNullException(nameof(options.Job));
+
+        if (options.IntervalMilliseconds <= 0)
```

Comment:
If Interval set to 0 but PushAfterBlock=true then this exception does not make sense?